src/util/image_path.cpp

```diff
@@ -0,0 +1,63 @@
+#include "filesys.h"
+#include "util/image_path.h"
+#include "util/string.h"
+#include <string>
+
+namespace {
+	/// An array of image extensions ordered roughly by commonness.
+	constexpr const char *image_extensions[]{
+		"png", "jgp", "bmp", "tga", "pcx", "ppm", "psd", "wal", "rgb"};
+}
+
+/*
+	Replaces the filename extension.
+	eg:
+		std::string image = "a/image.png"
+		replace_ext(image, "jpg")
+		-> image = "a/image.jpg"
+	Returns true on success.
+*/
+static bool replace_ext(std::string &path, const char *ext)
+{
+	if (ext == nullptr)
+		return false;
+	// Find place of last dot, fail if \ or / found.
+	int last_dot_i = -1;
+	for (int i = path.size() - 1; i >= 0; --i) {
+		if (path[i] == '.') {
+			last_dot_i = i;
+			break;
+		}
+
+		if (path[i] == '\\' || path[i] == '/')
+			break;
+	}
```

Comment:
std::string::find_last_of?

Reply:
I'm not very familiar with STL and I've never heard of this. Thank you for suggesting it, and I'll definitely check the docs on it and see whether I can apply it here.